extender/arpSyndicateSubdomainDiscovery.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * This script uses the API of ARPSyndicate's Subdomain Center (https://www.subdomain.center/) to
+ * find and add subdomains to the Sites Tree. When it is enabled, it runs automatically for each 
+ * new domain added to the Sites Tree.
+ */
+
+const HistoryReference = Java.type("org.parosproxy.paros.model.HistoryReference")
+const HttpSender = Java.type("org.parosproxy.paros.network.HttpSender")
+const HttpMessage = Java.type("org.parosproxy.paros.network.HttpMessage")
+const URI = Java.type("org.apache.commons.httpclient.URI")
+const requestedSubdomains = []
+
+function consumer(event) {
+    if (event.getEventType() != "site.added") return
+    try {
+        const siteNode = event.getTarget().getStartNode()
+        const host = siteNode.getHistoryReference().getURI().getHost()    
+        if (requestedSubdomains.indexOf(host) != -1) {
+            // Don't run for subdomain nodes created by this script
+            return
+        }
+        const apiUri = new URI(`https://api.subdomain.center/?domain=${host}`, true)
+        const apiMsg = new HttpMessage(apiUri)
+        const sender = new HttpSender(model.getOptionsParam().getConnectionParam(), true, HttpSender.MANUAL_REQUEST_INITIATOR)
```

Comment:
- Could be instantiated just once (along `requestedSubdomains`?).
- Should use int constructor (i.e. `new HttpSender(HttpSender.MANUAL_REQUEST_INITIATOR)`).